metricbeat/module/windows/perfmon/reader.go

```diff
@@ -124,6 +129,44 @@ func (re *Reader) Read() ([]mb.Event, error) {
 	return events, nil
 }
 
+func (re *Reader) getValues() (map[string][]pdh.CounterValue, error) {
+	var interval uint32 = 10
+	var val map[string][]pdh.CounterValue
+	rand.Seed(time.Now().UnixNano())
+	title := windows.StringToUTF16Ptr(randSeq(10))
+	event, err := windows.CreateEvent(nil, 0, 0, title)
+	if err != nil {
+		return nil, err
+	}
+	err = re.query.CollectDataEx(interval, event)
+	if err != nil {
+		return nil, err
+	}
+	waitfor, err := windows.WaitForSingleObject(event, windows.INFINITE)
+	if err != nil {
+		return nil, err
+	}
+	if waitfor == windows.WAIT_OBJECT_0 {
+		val, err = re.query.GetFormattedCounterValues()
+		if err != nil {
+			return nil, err
+		}
+	} else if waitfor == windows.WAIT_FAILED {
+		return nil, errors.New("WaitForSingleObject has failed")
+	}
```

Comment:
nit: this could be a switch clause, perhaps handling any other unexpected value

Reply:
changed it to switch